src/app/helpers/page/pageInfo.ts

```diff
@@ -6,45 +6,50 @@ import {
   Menus,
   ResolverList,
 } from "src/app/interfaces/menusInterfaces";
+import { PageComponent } from "./pageComponent";
 
 /**
  * Page info interface.
  * This stores information required to generate the various menus of the page.
  * Also stores metadata about the page, like the icon to use, and the route
  * for that page.
- * @extends MenuItem
  * @extends Data
  */
-export interface PageInfoInterface extends Data {
-  self: MenuRoute;
+export interface IPageInfo extends Data {
   category: Category;
-  menus: Menus;
+  self?: MenuRoute & { pageComponent?: Type<PageComponent> };
```

Comment:
`self` isn't a super descriptive name for this parameter. Is there a convention for this, or would another name be more suited such as `menuItem`? At minimum, this parameter should have a comment explaining its purpose.

Reply:
yeah I didn't like self either, but it was what was there.